tests/testthat/test-posting.R

```diff
@@ -0,0 +1,58 @@
+# this one is bombing on devtools::check because it thinks there's no webhook set up (there is).
+# it also weirdly works fine when I click "Run Tests" in RStudio.
+# test_that("slackr_bot posts", {
+#   slackr_bot_test <- slackr_bot('testing slackr_bot', incoming_webhook_url = Sys.getenv("SLACK_INCOMING_URL_PREFIX"))
+#   expect_equal(rawToChar(slackr_bot_test$content), 'ok')
+# })
+
+test_that("slackr posts", {
+  slackr_msg_test <- slackr('Testing')
+  expect_equal(content(slackr_msg_test)$ok, TRUE)
+
+  slackr_tbl_test <- slackr(head(iris))
+  expect_equal(content(slackr_tbl_test)$ok, TRUE)
+
+  slackr_lm_test <- slackr(
+    summary(lm(Petal.Width ~ Sepal.Width, data = iris))
+  )
+  expect_equal(content(slackr_lm_test)$ok, TRUE)
+})
+
+test_that("ggslackr posts", {
+  slackr_gg_test <- ggslackr(
+    ggplot(data = iris, aes(x = Petal.Width, y = Petal.Length, color = Species)) +
+      geom_point()
+  )
+  expect_equal(content(slackr_gg_test)$ok, TRUE)
+})
+
+test_that("slackr_msg posts", {
+  slackr_msg_test <- slackr_msg('Testing')
+  expect_equal(content(slackr_msg_test)$ok, TRUE)
+})
+
+test_that("text_slackr posts", {
+  text_slackr_test <- text_slackr('Testing')
+  expect_equal(content(text_slackr_test)$ok, TRUE)
+})
+
+test_that("slackr_delete works", {
+  slackr_msg('Testing deletion')
+  slackr_delete_test <- slackr_delete(1)
+  expect_equal(content(slackr_delete_test[[1]])$ok, TRUE)
+})
+
+test_that("slackr_upload posts", {
+  save(x, file = 'slackr_upload_test.Rdata')
```

Comment:
This needs to write to a tempfile, not the working directory

Reply:
good catch. will fix this